server/pbench/bin/pbench-server-prep-shim-002

```diff
@@ -0,0 +1,175 @@
+> $status
+results_dir=$(getconf.py receive-dir-002 pbench-server)
+# Check for results that are ready for processing: version 002 agents
+# upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+# after they are done with MD5 checking so that's what we look for.
+find ${results_dir} -maxdepth 2 -name *.tar.xz.md5 > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read ck ;do
+    # full pathname of tarball
+    tb=${ck%.md5}
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # the controller hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    dest=${archive}/${controller}
+
+    # sanity checks
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" >&4
+        if [ -f ${tb}.md5 ] ;then
+            quarantine ${quarantine}/${controller} ${tb}.md5
+        fi
+        nquarantined=$nquarantined+1
+        nerrs=$nerrs+1
+        continue
+    fi
+    # Impossible: the md5 file disappeared between the find and this check!
+    if [ ! -f ${tb}.md5 ] ;then
+        echo "$TS: ${tb}.md5 file does not exist" >&4
+        quarantine ${quarantine}/${controller} ${tb}
+        nquarantined=$nquarantined+1
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    pushd $(dirname $tb) >/dev/null 2>&4
+    md5sum --check $(basename $tb).md5
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: ${tb} failed MD5 check" >&4
+        quarantine ${quarantine}/${controller} ${tb} ${tb}.md5
+        nquarantined=$nquarantined+1
+        nerrs=$nerrs+1
+        popd >/dev/null 2>&4
+        continue
+    fi
+    popd > /dev/null 2>&4
+
+    if [ -f ${dest}/${resultname}.tar.xz -o -f ${dest}/${resultname}.tar.xz.md5 ] ;then
+        echo "$TS: ${tb} duplicate name" >&4
+        quarantine ${duplicates}/${controller}  ${tb} ${tb}.md5
+        ndups=$ndups+1
+        nerrs=$nerrs+1
+    fi
+
+    mkdir -p ${dest}/TODO
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"mkdir -p ${dest}\", status $sts" |
+            tee -a $status >&4
+        quarantine ${errors}/${controller} ${tb} ${tb}.md5
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    mv ${tb} ${tb}.md5 $dest
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"mv ${tb} ${tb}.md5 $dest\", status $sts" |
+            tee -a $status >&4
+        quarantine ${errors}/${controller} ${tb} ${tb}.md5
+        nerrs=$nerrs+1
+    fi
+
+    pushd $dest/TODO > /dev/null >/dev/null 2>&1
+    ln -s $dest/$(basename $tb) .
+    sts=$?
+    if [ $sts -ne 0 ] ;then
+        echo "$TS: Failed: \"ln -s $dest/$(basename $tb) .\", status $sts" |
+            tee -a $status >&4
+        quarantine ${errors}/${controller} ${dest}/${tb} ${dest}/${tb}.md5 ${prefix}
```

Comment:
Should we count the number of times we quarantined data?

Reply:
This counts as an error only.